Add support for Siege versions older than 3.0.8

florinpapa opened the pull request:
Older Siege versions than 3.0.8 use the HTTP/1.0 protocol instead of HTTP/1.1, causing the run-siege script to parse the output incorrectly. Changed the way parsing is done to solve this issue and output correct information for HTTP/1.0 as well.

cooperlees commented:
Do we really want to measure using HTTP 1.0? Not many clients out there today will access IG over HTTP 1.0 ...

florinpapa commented:
It is true that HTTP/1.0 will not be seen in production very often, but someone running the Django Workload using an older Siege version (2.78) will not be able to see the percentile and the URL hit percentages because the run-siege script looks for "HTTP/1.1" explicitly in the logs.

I can add a warning message if the protocol used is HTTP/1.0, does that sound ok?

cooperlees commented:
Why can't you just always use a newer siege that can do HTTP 1.1?

If that is painful for now, lets add a warning and I'll merge.

florinpapa commented:
The client/README.md file does say that the expected Siege version is 3.0.8, but someone who tries to run the workload might miss this detail. I think we should at least add a warning when HTTP/1.0 is used, so that the user knows why the percentiles and URL hit percentages could not be computed.

florinpapa commented:
Added a new commit that prints a warning if HTTP/1.0 is detected. Reverted the previous changes to make sure the user uses HTTP/1.1. Sample output below:

florin@flubuntu:~/django_wl//client$ DURATION=5S ./run-siege --single
Running iteration 1 --- DONE
WARNING: Got 265 HTTP/1.0 requests. This workload is intended to use HTTP/1.1, therefore some metrics will not not be computed. Please change the "protocol" variable in siegerc to HTTP/1.1 or install a newer Siege
Please see full Siege log in /tmp/siege_out_1


URL hit percentages:
/bundle_tray:		0.0%, expected 25%
/timeline:		0.0%, expected 25%
/feed_timeline:		0.0%, expected 26%
/seen:			0.0%, expected 5%
/inbox:			0.0%, expected 19%

Transactions:				265.0 hits ---- RSD 0.0
Availability:				100.0 % ---- RSD 0.0
Elapsed time:				4.5 secs ---- RSD 0.0
Data transferred:			0.42 MB ---- RSD 0.0
Response time:				2.18 secs ---- RSD 0.0
Transaction rate:			58.89 trans/sec ---- RSD 0.0
Throughput:				0.09 MB/sec ---- RSD 0.0
Concurrency:				128.26  ---- RSD 0.0
Successful transactions:		265.0  ---- RSD 0.0
Failed transactions:			0.0  ---- RSD 0
Longest transaction:			3.66  ---- RSD 0.0
Shortest transaction:			0.17  ---- RSD 0.0
P50:					N/A, please check Siege output file(s)
P90:					N/A, please check Siege output file(s)
P95:					N/A, please check Siege output file(s)
P99:					N/A, please check Siege output file(s)

Full Siege output is available in /tmp/siege_out_[N]

cooperlees commented:
Thanks!
